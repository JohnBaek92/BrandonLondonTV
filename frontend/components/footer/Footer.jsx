import React from "react";
import { Link } from "react-router-dom";

class Footer extends React.PureComponent {
  constructor() {
    super();
    this.state = { image: window.images.circle_logo };

    this.changeImage = this.changeImage.bind(this);
    this.changeBack = this.changeBack.bind(this);
  }

  changeImage() {
    this.setState({ image: window.images.circle_logo2 });
  }

  changeBack() {
    this.setState({ image: window.images.circle_logo });
  }

  render() {
    return (
      <footer className="footer">
        <div className="container">
          <nav>
            <div className="nav-left navbar-item" />
            <div className="nav-center middle-centered">
              <div className="social-tag">
                <a
                  className="social-tag"
                  href="https://www.facebook.com/BrandonLondonTv/"
                  target="_blank"
                >
                  <i className="fa fa-3x fa-facebook" />
                </a>
              </div>
              <div className="social-tag">
                <a
                  className="social-tag"
                  href="https://twitter.com/brandonlondontv"
                  target="_blank"
                >
                  <i className="fa fa-3x fa-twitter" />
                </a>
              </div>
              <div id="footer-logo">
                <Link
                  to="/"
                  onMouseEnter={this.changeImage}
                  onMouseLeave={this.changeBack}
                >
                  <img className="footer-logo" src={this.state.image} />
                </Link>
              </div>
              <div className="social-tag">
                <a
                  className="social-tag"
                  href="https://www.youtube.com/user/TheCulturedAthlete"
                  target="_blank"
                >
                  <i className="fa fa-3x fa-youtube" />
                </a>
              </div>
              <div className="social-tag">
                <a
                  className="social-tag"
                  href="https://www.instagram.com/brandonlondontv/"
                  target="_blank"
                >
                  <i className="fa fa-3x fa-instagram" />
                </a>
              </div>
            </div>
            <div className="nav-right navbar-item" />
          </nav>
        </div>
      </footer>
    );
  }
}

export default Footer;
