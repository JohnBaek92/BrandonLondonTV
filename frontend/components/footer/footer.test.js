import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Footer from "./Footer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { MemoryRouter } from "react-router";

var jsdom = require("jsdom");

configure({ adapter: new Adapter() });

describe("Footer component should render as expected", () => {
  const { JSDOM } = jsdom;

  let { document } = new JSDOM("").window;
  let images = { circle_logo: "foo", circle_logo2: "foo" };
  global.window = {
    ...document
  };
  global.window.images = images;
  it("Renders the Footer correctly", () => {
    const component = mount(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );
    const tree = toJson(component);

    expect(tree).toMatchSnapshot();
  });
});
