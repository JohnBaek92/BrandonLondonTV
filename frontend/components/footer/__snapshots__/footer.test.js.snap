// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render as expected Renders the Footer correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "ospptr",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "ospptr",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Footer>
      <footer
        className="footer"
      >
        <div
          className="container"
        >
          <nav>
            <div
              className="nav-left navbar-item"
            />
            <div
              className="nav-center middle-centered"
            >
              <div
                className="social-tag"
              >
                <a
                  className="social-tag"
                  href="https://www.facebook.com/BrandonLondonTv/"
                  target="_blank"
                >
                  <i
                    className="fa fa-3x fa-facebook"
                  />
                </a>
              </div>
              <div
                className="social-tag"
              >
                <a
                  className="social-tag"
                  href="https://twitter.com/brandonlondontv"
                  target="_blank"
                >
                  <i
                    className="fa fa-3x fa-twitter"
                  />
                </a>
              </div>
              <div
                id="footer-logo"
              >
                <Link
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  replace={false}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <img
                      className="footer-logo"
                      src="foo"
                    />
                  </a>
                </Link>
              </div>
              <div
                className="social-tag"
              >
                <a
                  className="social-tag"
                  href="https://www.youtube.com/user/TheCulturedAthlete"
                  target="_blank"
                >
                  <i
                    className="fa fa-3x fa-youtube"
                  />
                </a>
              </div>
              <div
                className="social-tag"
              >
                <a
                  className="social-tag"
                  href="https://www.instagram.com/brandonlondontv/"
                  target="_blank"
                >
                  <i
                    className="fa fa-3x fa-instagram"
                  />
                </a>
              </div>
            </div>
            <div
              className="nav-right navbar-item"
            />
          </nav>
        </div>
      </footer>
    </Footer>
  </Router>
</MemoryRouter>
`;
